src/movies.js

```diff
@@ -1,25 +1,168 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+  const directors = moviesArray.map((movie) => movie.director);
+  return directors;
+}
+
+// BONUS ITERATION (with the filter for duplicates):
+function getAllDirectorsBonus(moviesArray) {
+  const directors = moviesArray.reduce((directors, movie) => {
+    if (!directors.includes(movie.director)) {
+      directors.push(movie.director);
+    }
+    return directors;
+  }, []);
+  return directors;
+}
```

Comment:
Also another way to do it its with this method 

```suggestion
// BONUS ITERATION (with the filter for duplicates):
function getAllDirectorsBonus(moviesArray) {
 return [...new Set(moviesArray)]
}
```